model_one/cli.py

```diff
@@ -1,198 +1,274 @@
+        method, url = cls.V0["status"]
+
+        return cls._request(method, url.format(id))
+
+    @classmethod
+    def fit(cls, id: str) -> dict:
+        method, url = cls.V0["fit"]
+
+        return cls._request(method, url.format(id))
+
+
+class BeyondmlModelStatus(Enum, str):
+    READY = "ready"
+    CREATED = "created"
+    DATASETS_LOADED = "datasetsloaded"
+
+
+class BeyondmlModelType(Enum, str):
+    GENERATOR = "generator"
+    CLASSIFIER = "classifier"
+
+
+def inited(m: callable):
+    @wraps(m)
+    def _wrapper(self: BeyondmlModel, *args, **kwargs):
+        if not self.is_inited:
+            raise BeyondmlModelException("Initialize the model")
+        
+        return m(self, *args, **kwargs)
+    
+    return _wrapper
+
+
+class BeyondmlModel():
+    _id: str = None
+    _status: str = None
+    _model_type: str = None
+
+    def __init__(self, model_name: str, status: str, model_type: str, *args, **kwargs):
+        self._id = model_name
+        self._status = status
+        self._model_type = model_type
+    
+    @classmethod
+    def from_dict(cls, model: dict) -> 'BeyondmlModel':
+        return cls(**model)
+    
+    @classmethod
+    def from_id(cls, id: str) -> 'BeyondmlModel':
+        r = BeyondmlAPI.status(id)
+        return cls.from_dict(r)
+    
+    @classmethod
+    def create(cls, data: dict) -> 'BeyondmlModel':
+        r = BeyondmlAPI.create(data)
+        return cls.from_dict(r)
+
+    @classmethod
+    def create_classifier(cls, train_iters: int = None):
+        model = {"model_type": "classifier"}
+
+        if train_iters:
+            model["model_params"] = {"train_iters": train_iters}
+        
+        return cls.create(model)
+
+    @classmethod
+    def create_generator(cls, train_iters: int = None):
+        model = {"model_type": "generator"}
+
+        if train_iters:
+            model["model_params"] = {"train_iters": train_iters}
+        
+        return cls.create(model)
+
+    @classmethod
+    def models(cls) -> List['BeyondmlModel']:
+        r = BeyondmlAPI.models()
+
+        return [
+            cls.from_dict(data) for data in r.get("models", [])
+        ]
+
+    @classmethod
+    def load(cls, filename: str) -> 'BeyondmlModel':
+        with open(filename, "r") as fl:
+            data = json.load(fl)
+            return cls.from_dict(data)
+
+    @inited
     def save(self, filename):
-        with open(filename, 'w') as fl:
+        with open(filename, "w") as fl:
             json.dump({
-                'model_name': self._id,
-                'status': self._status,
-                'model_type': self._model_type,
+                "model_name": self._id,
+                "status": self._status,
+                "model_type": self._model_type,
             }, fl)
 
-    def _upload_fit(self, train_X, train_y, validate_X, validate_y):
-        self.status()
-        if self._status == 'Created':
+    @property
+    def is_initied(self) -> bool:
```

Comment:
properties allow us to use methods with no arguments besides `self` as object fields
methods like "is_%something%" are usually good place for properties